src/resmom/start_exec.c

```diff
@@ -2931,6 +2952,34 @@ finish_exec(job *pjob)
 	vnl_t			*vnl_fails = NULL;
 	vnl_t			*vnl_good = NULL;
 
+	if (mock_run) {
+		resource_def *rd;
+		attribute *wallt;
+		resource *wall_req;
+		int walltime = 0;
+
+		rd = find_resc_def(svr_resc_def, "walltime", svr_resc_size);
+		wallt = &pjob->ji_wattr[(int)JOB_ATR_resource];
+		wall_req = find_resc_entry(wallt, rd);
+		if (wall_req != NULL) {
+			walltime = wall_req->rs_value.at_val.at_long;
+			start_walltime(pjob);
+		}
+
+		time_now = time(NULL);
+
+		/* Add a work task that runs when the job is supposed to end */
+		set_task(WORK_Timed, time_now + walltime, mock_run_end_job_task, pjob);
+
+		sprintf(log_buffer, "Started mock run of job");
+		log_event(PBSEVENT_JOB, PBS_EVENTCLASS_JOB,
+			LOG_INFO, pjob->ji_qs.ji_jobid, log_buffer);
+
+		mom_set_use(pjob);
```

Comment:
I don't see a change in mom_set_use, how will you ensure that resources_used are correctly set?